src/main/java/co/cask/gcp/bigquery/BigQueryDelete.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.gcp.bigquery;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Macro;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.etl.api.action.Action;
+import co.cask.cdap.etl.api.action.ActionContext;
+import co.cask.gcp.common.GCPConfig;
+import com.google.cloud.ServiceOptions;
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.DatasetId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+
+/**
+ * This class <code>BigQueryDelete</code> deletes a dataset or a table.
+ * <p>
+ * If the table or dataset does not exist, the delete action is still successful.
+ */
+@Plugin(type = Action.PLUGIN_TYPE)
+@Name(BigQueryDelete.NAME)
+@Description("Delete a Google BigQuery dataset or table.")
+public final class BigQueryDelete extends Action {
```

Comment:
let's rename this to BigQueryDrop instead of BigQueryDelete. Feedback from users is that in the SQL world, delete gives the impression that only the data will be removed, but the table will stay (``DELETE from``). However, this deletes the table itself, which is equivalent to the ``DROP`` statement in SQL